cmd/self-update.go

```diff
@@ -27,26 +30,40 @@ func init() {
 func NewKoolSelfUpdate() *KoolSelfUpdate {
 	return &KoolSelfUpdate{
 		*newDefaultKoolService(),
+		task.NewRunner(),
 		&updater.DefaultUpdater{RootCommand: rootCmd},
 	}
 }
 
 // Execute runs the self-update logic with incoming arguments.
 func (s *KoolSelfUpdate) Execute(args []string) (err error) {
-	var latestVersion semver.Version
+	newVersion, err := s.taskRunner.Run("Updating kool version", func() (taskResult interface{}, taskError error) {
```

Comment:
Having the return `taskResult interface{}` is a bit odd, I would rather try to make this in a way the task.Runner would wrap cmd.KoolService and encapsulate the cmd.KooService.Execute call with proper outout/error handling.